src/sunadaptcontroller/imexgus/sunadaptcontroller_imexgus.c

```diff
@@ -98,16 +97,17 @@ SUNAdaptController SUNAdaptController_ImExGus(SUNContext sunctx)
  * Function to set ImExGus parameters
  */
 
-int SUNAdaptController_SetParams_ImExGus(SUNAdaptController C, sunrealtype k1e,
-                                         sunrealtype k2e, sunrealtype k1i,
-                                         sunrealtype k2i)
+SUNErrCode SUNAdaptController_SetParams_ImExGus(SUNAdaptController C,
+                                                sunrealtype k1e, sunrealtype k2e,
+                                                sunrealtype k1i, sunrealtype k2i)
 {
-  if (C == NULL) { return SUNADAPTCONTROLLER_ILL_INPUT; }
+  SUNFunctionBegin(C->sunctx);
+  SUNAssert(C, SUN_ERR_ARG_CORRUPT);
```

Comment:
Does this order make sense?  If `C` is `NULL` (tested in the second line) then the first line would segfault.  This pattern repeats below.

I'd think that the second line could just be removed, since it is guaranteed to succeed if it is ever executed.

Reply:
You are right, the second line does not make sense to have